core/errors/error-match-error.ts

```diff
@@ -2,7 +2,7 @@ import { MatchError } from "./match-error";
 
 export class ErrorMatchError extends MatchError {
 
-   public constructor(actualError: Error,
+   public constructor(actualError: Error | undefined | null,
```

Comment:
Should this be the union type everywhere? Perhaps it should be cast instead at some point?